File: HospitalDuty.Domain/Entities/Shift.cs
using System;
using HospitalDuty.Domain.Enums;

namespace HospitalDuty.Domain.Entities;

public class Shift
{
    public int Id { get; set; }

    public Guid DepartmentId { get; set; }
    public Department Department { get; set; } = default!;

    public Guid EmployeeId { get; set; }
    public Employee Employee { get; set; } = default!;

    public Guid HospitalId { get; set; }
    public Hospital Hospital { get; set; } = default!;

    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }

    public ShiftType ShiftType { get; set; } = ShiftType.Normal;

    public string Notes { get; set; } = string.Empty;
}

File: client/src/types/shift.ts
export const ShiftType = {
  Normal: 0,
  Night: 1,
  Emergency: 2
} as const;

export type ShiftType = typeof ShiftType[keyof typeof ShiftType];

export interface Shift {
  id: string;
  employeeId: string;
  hospitalId: string;
  departmentId: string;
  startTime: string; // ISO string
  endTime: string; // ISO string
  shiftType: ShiftType;
  notes: string;
  employeeName: string;
  hospitalName: string;
  departmentName: string;
}

export interface CreateShiftRequest {
  employeeId: string;
  hospitalId: string;
  departmentId: string;
  startTime: string; // ISO string
  endTime: string; // ISO string
  shiftType: ShiftType;
  notes: string;
}

export interface UpdateShiftRequest {
  employeeId: string;
  hospitalId: string;
  departmentId: string;
  startTime: string; // ISO string
  endTime: string; // ISO string
  shiftType: ShiftType;
  notes: string;
}

export interface ShiftResponse {
  success: boolean;
  message: string;
  data?: Shift[];
}
